app/uk/gov/hmrc/economiccrimelevyreturns/controllers/CheckYourAnswersController.scala

```diff
@@ -195,21 +206,87 @@ class CheckYourAnswersController @Inject() (
   private def base64EncodeHtmlView(html: String): String = Base64.getEncoder
     .encodeToString(html.getBytes)
 
-  private def createAndEncodeHtmlForPdf()(implicit request: ReturnDataRequest[_]): String = {
-    val date         = LocalDate.now
-    val organisation = eclDetails()
-    val contact      = contactDetails()
-    val amendReason  = amendReasonDetails()
-    base64EncodeHtmlView(
-      amendReturnPdfView(
-        ViewUtils.formatLocalDate(date),
-        organisation.copy(rows = organisation.rows.map(_.copy(actions = None))),
-        contact.copy(rows = contact.rows.map(_.copy(actions = None))),
-        amendReason.copy(
-          rows = amendReason.rows.map(_.copy(actions = None)),
-          attributes = Map("id" -> "amendReason")
-        )
-      ).toString()
+  private def viewHtmlOrError()(implicit
+    request: ReturnDataRequest[AnyContent]
+  ): EitherT[Future, ResponseError, HtmlFormat.Appendable] =
+    EitherT {
+      request.eclReturn.returnType match {
+        case Some(FirstTimeReturn) => Future.successful(Right(view(viewModelWithoutEclReturnSubmission(), appConfig)))
+        case Some(AmendReturn)     =>
+          if (appConfig.getEclReturnEnabled) {
+            request.periodKey match {
+              case None            =>
+                Future.successful(Left(ResponseError.internalServiceError("Unable to find period key")))
+              case Some(periodKey) =>
+                viewModelWithEclReturnSubmission(periodKey).fold(
+                  error => Left(error),
+                  viewModel => Right(view(viewModel, appConfig))
+                )
+            }
+          } else {
+            Future.successful(Right(view(viewModelWithoutEclReturnSubmission(), appConfig)))
+          }
+        case None                  => Future.successful(Left(ResponseError.internalServiceError("Unable to find return type")))
+      }
+    }
+
+  private def pdfViewModelWithEclReturnSubmission(periodKey: String)(implicit
+    hc: HeaderCarrier,
+    request: ReturnDataRequest[AnyContent]
+  ): EitherT[Future, ResponseError, AmendReturnPdfViewModel] =
+    EitherT {
+      (for {
+        eclReturnSubmission <-
+          returnsService.getEclReturnSubmission(periodKey, request.eclRegistrationReference).asResponseError
```

Comment:
you can just map instead of folding ```
returnsService.getEclReturnSubmission(periodKey, request.eclRegistrationReference).asResponseError.map(eclReturnSubmission =>
           Right(
             AmendReturnPdfViewModel(
               date = LocalDate.now(),
               eclReturn = request.eclReturn,
               eclReturnSubmission = Some(eclReturnSubmission)
             )
           ))
```

Reply:
Done